spec/support/shared_examples/password_strength.rb

```diff
@@ -94,4 +94,26 @@
     end
     expect(form.submit(password: password)).to eq result
   end
+
+  it 'does not allow a password that does not have the minimum number of graphemes' do
+    user = build_stubbed(:user, email: 'custom@example.com', uuid: '123')
+    allow(user).to receive(:reset_password_period_valid?).and_return(true)
+    form = form_class.constantize.new(user)
+    password = 'a7K!hf🇺🇸🇺🇸🇺🇸'
+    errors = {
+      password: [t('errors.messages.too_short', count: 12)],
+    }
+    extra = hash_including(user_id: '123')
+    result = instance_double(FormResponse)
+
+    if %w[PasswordForm ResetPasswordForm].include?(form_class)
+      expect(FormResponse).to receive(:new).
+        with(success: false, errors: errors, extra: extra).and_return(result)
+    else
+      expect(FormResponse).to receive(:new).
+        with(success: false, errors: errors).and_return(result)
+    end
+
+    expect(form.submit(password: password)).to eq result
```

Comment:
I really dislike this pattern of stubbing `:new`... especially on a value object.... can we just expect on the results?

```suggestion
    result = form.submit(password: password)

    expect(result.success?).to eq(false)
    expect(result.errors).to eq(errors)
    expect(result.extra).to include(user_id: '123') if result.extra.present?
```

Reply:
Strong agree so I changed a bunch of them https://github.com/18F/identity-idp/pull/4955/commits/f4fa0c333ddd1fd23511b26d72925ed65bd2bcd4

Reply:
Thank you! ❤️